examples/structured_data/classification_with_grn_and_vsn.py

```diff
@@ -0,0 +1,413 @@
+"""
+Title: Classification with Gated Residual and Variable Selection Networks
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/01/15
+Last modified: 2021/01/15
+Description: Using Gated Residual and Variable Selection Networks for income level prediction.
+"""
+
+"""
+## Introduction
+
+This example demonstrates the use of Gated
+Residual Network (GRN) and Variable Selection Networks (VSN), proposed by
+[Bryan Lim et al.](https://arxiv.org/abs/1912.09363), for structured data
+classification. GRNs give the flexibility to the model to apply non-linear
+processing only where needed. VSNs allow the model to softly remove any
+unnecessary noisy inputs which could negatively impact performance.
+Together, those techniques help improving the learning capacity of deep neural
+network models.
+
+
+Note that this example should be run with TensorFlow 2.3 or higher.
+"""
+
+"""
+## The dataset
+
+This example uses the [United States Census Income
+Dataset](https://archive.ics.uci.edu/ml/datasets/Census-Income+%28KDD%29) provided
+by the [UC Irvine Machine Learning Repository](https://archive.ics.uci.edu/ml/index.php).
+The task is binary classification to determine whether a person makes over 50K a year.
+
+The dataset includes ~300K instances with 41 input features, 7 of which numerical,
+and the other 34 are categorical.
+"""
+
+"""
+## Setup
+"""
+
+import math
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+"""
+## Prepare the data
+
+First we load the data from the UCI Machine Learning Repository to a pandas Dataframe.
```

Comment:
Suggestion:

- Lint everything before submitting the PR. It's "Pandas" not "pandas".

```suggestion
First we load the data from the UCI Machine Learning Repository to a Pandas Dataframe.
```